pkg/logging/logging.go

```diff
@@ -41,9 +41,52 @@ const (
 var loggingStderr bool
 var loggingW io.Writer
 var loggingLevel Level
+var logger *lumberjack.Logger
 
 const defaultTimestampFormat = time.RFC3339
 
+// LogOptions specifies the configuration of the log
+type LogOptions struct {
+	MaxAge     *int  `json:"maxAge,omitempty"`
+	MaxSize    *int  `json:"maxSize,omitempty"`
+	MaxBackups *int  `json:"maxBackups,omitempty"`
+	Compress   *bool `json:"compress,omitempty"`
+}
+
+// SetLogOptions set the LoggingOptions of NetConf
+func SetLogOptions(options *LogOptions) {
+	if options != nil {
+		if options.MaxAge != nil {
+			logger.MaxAge = *options.MaxAge
+		} else {
+			logger.MaxAge = 5
+		}
+		if options.MaxSize != nil {
+			logger.MaxSize = *options.MaxSize
+		} else {
+			logger.MaxSize = 100
+		}
+		if options.MaxBackups != nil {
+			logger.MaxBackups = *options.MaxBackups
+		} else {
+			logger.MaxBackups = 5
+		}
+		if options.Compress != nil {
+			logger.Compress = *options.Compress
+		} else {
+			logger.Compress = true
+		}
+	} else {
+		// give some default value
+		logger.MaxSize = 100
+		logger.MaxAge = 5
+		logger.MaxBackups = 5
+		logger.Compress = true
+	}
```

Comment:
This may be more simple.

```
func SetLogOptions(options *LogOptions) {
        // give some default value
        logger.MaxSize = 100
        logger.MaxAge = 5
        logger.MaxBackups = 5
        logger.Compress = true
        if options != nil {
                if options.MaxAge != nil {
                        logger.MaxAge = *options.MaxAge
                }
                if options.MaxSize != nil {
                        logger.MaxSize = *options.MaxSize
                }
                if options.MaxBackups != nil {
                        logger.MaxBackups = *options.MaxBackups
                }
                if options.Compress != nil {
                        logger.Compress = *options.Compress
                }
        }

        loggingW = logger
}
```

Reply:
Sure!  Thank you for your comment and I totally agree with your suggestion.